vdp/pipeline/v1beta/integration.proto

```diff
@@ -0,0 +1,242 @@
+  // Method defines how the connection is set up.
+  enum Method {
+    // Unspecified.
+    METHOD_UNSPECIFIED = 0;
+    // Key-value collection. The user is responsible of fetching the connection
+    // details from the 3rd party service.
+    METHOD_DICTIONARY = 1;
+    // Access token created via OAuth 2.0 authorization.
+    METHOD_OAUTH = 2;
+  }
+  // UUID-formatted unique identifier.
+  string uid = 1 [
+    (google.api.field_behavior) = OUTPUT_ONLY,
+    (google.api.field_behavior) = IMMUTABLE
+  ];
+  // ID.
+  string id = 2 [(google.api.field_behavior) = REQUIRED];
+  // ID of the namespace owning the connection.
+  string namespace_id = 3 [(google.api.field_behavior) = REQUIRED];
+  // Integration ID. It determines for which type of components can reference
+  // this connection.
+  string integration_id = 4 [(google.api.field_behavior) = REQUIRED];
+  // Integration title. This helps the console display the results grouped by
+  // integration ID without needing an extra call to fetch title by integration
+  // ID.
+  string integration_title = 5 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // The IDs of the pipelines that use this integration. All the pipelines will
+  // be within the same namespace as the connection.
+  // The length of the list will be capped to 100.
+  repeated string pipeline_ids = 6 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Connection method. It references the setup schema provided by the
+  // integration.
+  Method method = 7 [(google.api.field_behavior) = REQUIRED];
+  // Connection details. This field is required on creation, optional on view.
+  // When viewing the connection details, the setup values will be redacted.
+  google.protobuf.Struct setup = 8 [(google.api.field_behavior) = REQUIRED];
+  // View defines how the integration is presented. The `setup` field is only
+  // showed in the FULL view.
+  View view = 9  [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Creation timestamp.
+  google.protobuf.Timestamp create_time = 10 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Last update timestamp.
+  google.protobuf.Timestamp update_time = 11 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+}
+
+// ListNamespaceConnectionsRequest represents a request to list the connections
+// created by a namespace.
+message ListNamespaceConnectionsRequest {
+  // Namespace ID.
+  string namespace_id = 1 [(google.api.field_behavior) = REQUIRED];
+  // The maximum number of items to return. The default and cap values are 10
+  // and 100, respectively.
+  optional int32 page_size = 2 [(google.api.field_behavior) = OPTIONAL];
+  // Page number. By default, the first page will be returned.
+  optional int32 page = 3 [(google.api.field_behavior) = OPTIONAL];
+  // Filter can hold an [AIP-160](https://google.aip.dev/160)-compliant filter
+  // expression.
+  // The following filters are supported:
+  // - `integration_id`
+  // - `q` (fuzzy search on connection ID, integration title or vendor)
```

Comment:
@donch1989 I'm not sure this is achievable through a single filter parameter. Otherwise the frontend should concatenate (OR) filters on these fields.

Reply:
Do you mean filter by `integration_id` and `q` in the same time?

Reply:
No, my question is whether we can add a `q` parameter (or with other name if needed) that can perform a fuzzy search on connection ID, integration title and vendor simultaneously, e.g. `q='googl'`.

If this isn't possible, then we'll need to add one filter per field, e.g. `id='googl' OR 'integration_title='googl' OR vendor='googl'`

Reply:
Yeah, we can add the `q` parameter. We can determine how to handle the search with `q` ourselves. But currently, it require some hard-code in the db transpiler.